boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -80,13 +89,19 @@ docker-build: isclean
 	${CONTAINER_ENGINE} tag $(OPERATOR_IMAGE_URI) $(OPERATOR_IMAGE_URI_LATEST)
 
 .PHONY: docker-push
-docker-push:
-	${CONTAINER_ENGINE} push $(OPERATOR_IMAGE_URI)
-	${CONTAINER_ENGINE} push $(OPERATOR_IMAGE_URI_LATEST)
+docker-push: docker-login docker-build
+	${CONTAINER_ENGINE} --config=${CONTAINER_ENGINE_CONFIG_DIR} push ${OPERATOR_IMAGE_URI}
+	${CONTAINER_ENGINE} --config=${CONTAINER_ENGINE_CONFIG_DIR} push ${OPERATOR_IMAGE_URI_LATEST}
 
 .PHONY: push
 push: docker-push
 
+.PHONY: docker-login
```

Comment:
Is there a way to check whether already authenticated? If so, it would be preferable to do that, and only require/use the cred vars if necessary.

Reply:
The presence of docker configuration doesn't guarantee that it has a working set of credentials, so I don't see how this could be done easily.

On second thoughts, docker pushes must be protected as much as possible. They are one of the most delicate parts of the system and we know the consequences of bad pushes to both the catalog or the operator image... I don't think it is a bad thing that creds are always requested.